### PR TITLE
`maybe-type-hint-destructured-args`: add missing `ann->symbol` call

### PR DESCRIPTION
**Pull request by vemv**
## Brief

Fixes nedap/speced.def#84

## QA/review plan

* Green build
  * Note new build step, new `integration.reflection-test` ns
* Note how https://circleci.com/gh/nedap/speced.def/217 fails
  * It's a separate branch created just for proving the new step works.

## Author checklist

* [x] I have QAed the functionality
* [x] The PR has a reasonably reviewable size and a meaningful commit history
* [x] I have run the [branch formatter](https://github.com/nedap/formatting-stack/blob/332a419034ab46fad526a5592f4257353bd695b6/src/formatting_stack/branch_formatter.clj) and observed no new/significative warnings
* [x] The build passes
* [x] I have self-reviewed the PR prior to assignment
* Additionally, I have code-reviewed iteratively the PR considering the following aspects in isolation:
  * [x] Correctness
  * [x] Robustness (red paths, failure handling etc)
  * [ ] Modular design
  * [x] Test coverage
  * [ ] Spec coverage
  * [ ] Documentation
  * [ ] Security
  * [ ] Performance
  * [ ] Breaking API changes
  * [x] Cross-compatibility (Clojure/ClojureScript)

## Reviewer checklist

* [ ] I have checked out this branch and reviewed it locally, running it
* [ ] I have QAed the functionality
* [ ] I have reviewed the PR
* Additionally, I have code-reviewed iteratively the PR considering the following aspects in isolation:
  * [ ] Correctness
  * [ ] Robustness (red paths, failure handling etc)
  * [ ] Modular design
  * [ ] Test coverage
  * [ ] Spec coverage
  * [ ] Documentation
  * [ ] Security
  * [ ] Performance
  * [ ] Breaking API changes
  * [ ] Cross-compatibility (Clojure/ClojureScript)

### CODE REVIEW

File: project.clj [hunk 1]
```diff
@@ -54,36 +54,38 @@
...
+                                                  :exclusions [com.cognitect/transit-clj
+                                                               com.google.code.findbugs/jsr305
+                                                               com.google.errorprone/error_prone_annotations]]
+                                                 [com.cognitect/transit-clj "0.8.313" #_"transitive"]
+                                                 [com.google.errorprone/error_prone_annotations "2.1.3" #_"transitive"]
+                                                 [com.google.code.findbugs/jsr305 "3.0.2" #_"transitive"]]}
+
+             :test               {:dependencies [[com.nedap.staffing-solutions/utils.test "1.5.0"]]}
+
+             :warn-on-reflection {:global-vars {*warn-on-reflection* true}}
```
Comment (thumbnail) [inline project.clj]: Should we enable this by default in !production ?

File: .circleci/config.yml [hunk 2]
```diff
@@ -42,6 +42,10 @@ jobs:
         name: 'Run JVM test suite'
         command: lein with-profile -dev test
 
+    - run:
+        name: 'Run JVM test suite, failing on reflection warnings'
+        command: lein with-profile -dev,+warn-on-reflection test 2>&1 | grep warning && exit 1 || exit 0
```
Comment (thumbnail) [inline .circleci/config.yml]: We could try to capture the output of the first test-run (L43), and only check the output in a similar way. That prevents the whole test-suite to run.

File: project.clj (hunk 1 above)
Comment (vemv) [reply project.clj]: Would love to, but in practice lots of dependencies (including the cljs compiler!) have unaddressed reflection warns, so this var would make dev startup far noisier.

File: .circleci/config.yml (hunk 2 above)
Comment (vemv) [reply .circleci/config.yml]: Yes, definitely. Such a setup could be even reused in all our projects.

Not immediately obvious how to achieve that though...

Two requirements:

* keep 'streaming' the output to Circle
* make it clear if the build failed due to red tests or due to grep

We can eventually give it a shot.

**Pull request merged**